dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -109,6 +84,16 @@
      */
     @Override
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+        // Optimistically avoid the queue in the fast path.
+        // Queuing adds contention between threads and should be avoided unless we need to shed load.
+        Optional<ListenableFuture<Response>> maybeResult = delegate.maybeExecute(endpoint, request);
+        if (maybeResult.isPresent()) {
+            ListenableFuture<Response> result = maybeResult.get();
+            numRunningRequests.incrementAndGet();
+            result.addListener(this::onCompletion, DIRECT);
+            return result;
```

Comment:
can we pull this logic out since it looks about the same as what's in `scheduleNextTask`?